### PR TITLE
Fix getting the image file modification date (+ introduce a way to get the file creation date, and break the API)

### PR DESCRIPTION
**Pull request by mz2**
Also, in the process optimises so that the value is not accessed multiple times from FS if it was genuinely missing or fetching it failed (might want to in the future still expose that failure state to the client too so it'd know to attempt to clear the cache to access disk again?).

### CODE REVIEW

**Comment by mz2** [comment]
I have tagged a beta release of "version 2.0.0" here because this introduces a breaking API change (the ambiguous `fileTimestamp` is separated into `fileModificationDate` and `fileCreationDate`).

https://github.com/mz2/Carpaccio/releases/tag/v2.0.0-beta1

File: Tests/CarpaccioTests/CarpaccioTests.swift [hunk 1]
```diff
@@ -202,11 +202,11 @@ class CarpaccioTests: XCTestCase {
         )
         XCTAssertEqual(imageMetadata.fNumber, (dictRep[ImageMetadata.CodingKeys.fNumber.dictionaryRepresentationKey] as! NSNumber).doubleValue)
         XCTAssertEqual(imageMetadata.exposureCompensation, (dictRep[ImageMetadata.CodingKeys.exposureCompensation.dictionaryRepresentationKey] as! NSNumber).doubleValue)
-        XCTAssertEqual(imageMetadata.flashMode, dictRep[ImageMetadata.CodingKeys.flashMode.dictionaryRepresentationKey] as! FlashMode)
```
Comment (mz2) [inline Tests/CarpaccioTests/CarpaccioTests.swift:205]: These changes were just because the linter complained about them.

**Comment by mz2** [comment]
Actually, I've removed the XCFramework building step since it was anyway not really working at least in my experience (the binary framework failed to be linked to with very cryptic compiler / linker errors that were not worth investigation time), since the build step has now bitrotten. Can be explored again when binary framework support in Xcode is confirmed alright for the purpose.

I also added some basic test coverage reporting here.

File: Sources/Carpaccio/Image.swift [hunk 2]
```diff
@@ -210,33 +211,65 @@ open class Image: Equatable, Hashable, CustomStringConvertible {
...
+        switch fileCreationDateAccessed {
+        case .successfully(let date):
+            return date
+        case .failedToAccess:
+            return nil
+        case .notAccessed:
+            guard let url = self.URL, url.isFileURL else {
+                return nil
+            }
+            if let fileTimestamp = try? FileManager.default.attributesOfItem(atPath: url.path)[.creationDate] as? Date {
```
Comment (mz2) [inline Sources/Carpaccio/Image.swift:254]: Dis is where the bugfix was done: `attributesOfFileSystem` -> `attributesOfItem`

**Review by markuspiipari — APPROVED** [review]

**Pull request merged**